src/core/update_initial_conditions.jl

```diff
@@ -95,33 +138,35 @@ function _make_initial_conditions!(
     parameters = model_has_parameters(psi_container)
     ic_container = get_initial_conditions(psi_container)
     if !has_initial_conditions(ic_container, key)
-        @debug "Setting $(key.ic_type) initial conditions for the status of all devices $(T) based on system data"
+        @debug "Setting $(key.ic_type) initial conditions for all devices $(T) based on system data"
         ini_conds = Vector{InitialCondition}(undef, length_devices)
         set_initial_conditions!(ic_container, key, ini_conds)
         for (ix, dev) in enumerate(devices)
             val_ = get_val_func(dev, key)
             val = parameters ? PJ.add_parameter(psi_container.JuMPmodel, val_) : val_
             ic = make_ic_func(ic_container, dev, val, cache)
             ini_conds[ix] = ic
-            @debug "set initial condition" key ic val
+            @debug "set initial condition" key ic val_
         end
     else
         ini_conds = get_initial_conditions(ic_container, key)
         ic_devices = Set((IS.get_uuid(ic.device) for ic in ini_conds))
         for dev in devices
             IS.get_uuid(dev) in ic_devices && continue
-            @debug "Setting $(key.ic_type) initial conditions for the status device $(PSY.get_name(dev)) based on system data"
-            val = get_val_func(dev, key)
+            @debug "Setting $(key.ic_type) initial conditions device $(PSY.get_name(dev)) based on system data"
+            val_ = get_val_func(dev, key)
+            val = parameters ? PJ.add_parameter(psi_container.JuMPmodel, val_) : val_
             ic = make_ic_func(ic_container, dev, val, cache)
             push!(ini_conds, ic)
-            @debug "set initial condition" key ic val
+            @debug "set initial condition" key ic val_
```

Comment:
You probably want this to be `val`, right?

Reply:
I want val_ because if the setting is to have parameters this will produce a `param_noname` entry instead of the value